### PR TITLE
Add internal schema

### PR DESCRIPTION
**Pull request by MadeByMike**
Adds a private schema.

Todo:

 - [x] Bikeshed name: "internal"
 - [x] Add some tests
 - [x] Should we block people from adding this to GraphQL app or AdminUI?
 - [x] Does this work with `executeQuery` without any other change? Previous testing required a schemaRouter for any multi-schema. Is this different?

### CODE REVIEW

**Comment by gautamsi** [comment]
is this aligned with future possibility of Apollo Federation (GraphQL gateway)?

File: packages/keystone/lib/ListTypes/list.js [hunk 1]
```diff
@@ -145,16 +145,20 @@ module.exports = class List {
...
-              (acc, [op, rule]) => ({ ...acc, [op]: !!rule }), // Reduce the entries to truthy values
-              {}
-            ),
-          }),
-          {}
-        ),
+        access: Object.entries(this.access)
+          .filter(([key]) => {
+            return key !== 'private';
+          })
```
Comment (timleslie) [inline packages/keystone/lib/ListTypes/list.js:151]:
```suggestion
          .filter(([key]) => key !== 'private')
```

File: packages/access-control/lib/access-control.js [hunk 2]
```diff
@@ -71,6 +75,7 @@ const parseAccess = ({ schemaNames, accessTypes, access, defaultAccess, parseAnd
             : defaultAccess
           : access
       ),
+      private: { ...accessTypes.reduce((acc, accessType) => ({ ...acc, [accessType]: true }), {}) },
```
Comment (timleslie) [inline packages/access-control/lib/access-control.js:78]:
```suggestion
      private: defaultObj(accessTypes, true),
```

**Comment by MadeByMike** [comment]
ToDo:
 - [ ] Rename 'private' to internal
 - [ ] Write a test that ties to use a GraphQL app with 'internal'

File: packages/app-admin-ui/index.js [hunk 3]
```diff
@@ -33,6 +33,12 @@ class AdminUIApp {
       throw new Error('Keystone 5 Admin currently only supports the `PasswordAuthStrategy`');
     }
 
+    if (schemaName === 'internal') {
+      throw new Error(
+        "The schemaName 'internal' is reserved name cannot be used in the AdminUIApp."
```
Comment (singhArmani) [inline packages/app-admin-ui/index.js:38]: Just a typo, `The schemaName 'internal' is **a** reserved name, **and** cannot be used in the AdminUIApp.`

File: packages/app-graphql/index.js [hunk 4]
```diff
@@ -10,6 +9,10 @@ class GraphQLApp {
     schemaName = 'public',
     apollo = {},
   } = {}) {
+    if (schemaName === 'internal') {
+      throw new Error("The schemaName 'internal' is reserved name cannot be used in a GraphQLApp.");
```
Comment (singhArmani) [inline packages/app-graphql/index.js:13]: Same typo as above. ☝️

File: packages/app-schema-router/README.md [hunk 5]
```diff
@@ -29,10 +29,10 @@ module.exports = {
     new GraphQLAppPlayground({ apiPath })
     new SchemaRouterApp({
       apiPath,
-      routerFn: (req) => req.session.keystoneItemId ? 'private' : 'public',
+      routerFn: (req) => req.session.keystoneItemId ? 'private' : 'admin',
```
Comment (singhArmani) [inline packages/app-schema-router/README.md:32]: As `private` is not a reserved name anymore, can we bring back the `private` now, and revert this ternary condition?

File: packages/app-admin-ui/index.js (hunk 3 above)
Comment (MadeByMike) [inline packages/app-admin-ui/index.js:38]:
```suggestion
        "The schemaName 'internal' is a reserved name cannot be used in the AdminUIApp."
```

File: packages/app-graphql/index.js (hunk 4 above)
Comment (MadeByMike) [inline packages/app-graphql/index.js:13]:
```suggestion
      throw new Error("The schemaName 'internal' is a reserved name cannot be used in a GraphQLApp.");
```

File: packages/app-admin-ui/index.js (hunk 3 above)
Comment (MadeByMike) [inline packages/app-admin-ui/index.js]:
```suggestion
        "The schemaName 'internal' is a reserved name cannot be used in the AdminUIApp."
```

File: api-tests/extend-graphql-schema/extend-graphql-schema.test.js [hunk 6]
```diff
@@ -46,11 +46,11 @@ multiAdapterRunners().map(({ runner, adapterName }) =>
         'Sets up access control properly',
         runner(setupKeystone, async ({ keystone }) => {
           expect(keystone._customProvider._extendedQueries.map(({ access }) => access)).toEqual([
-            { testing: true },
-            { testing: falseFn },
+            expect.objectContaining({ testing: true }),
```
Comment (timleslie) [inline api-tests/extend-graphql-schema/extend-graphql-schema.test.js:49]: Let's update this to explicitly contain the `internal` values as well.

File: packages/app-schema-router/README.md [hunk 7]
```diff
@@ -32,7 +32,7 @@ module.exports = {
       routerFn: (req) => req.session.keystoneItemId ? 'private' : 'public',
       apps: {
         public: new GraphQLApp({ apiPath, schemaName: 'public', graphiqlPath: undefined }),
-        private: new GraphQLApp({ apiPath, schemaName: 'private', graphiqlPath: undefined }),
+        admin: new GraphQLApp({ apiPath, schemaName: 'private', graphiqlPath: undefined }),
```
Comment (singhArmani) [inline packages/app-schema-router/README.md:35]:
```suggestion
        private: new GraphQLApp({ apiPath, schemaName: 'private', graphiqlPath: undefined }),
```

File: packages/access-control/lib/access-control.js [hunk 8]
```diff
@@ -79,6 +84,7 @@ const parseAccess = ({ schemaNames, accessTypes, access, defaultAccess, parseAnd
 module.exports = {
   parseCustomAccess({ defaultAccess, access = defaultAccess, schemaNames }) {
     const accessTypes = [];
+
```
Comment (timleslie) [inline packages/access-control/lib/access-control.js:87]:
```suggestion
```

File: .changeset/silver-walls-shake.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+---
```
Comment (timleslie) [inline .changeset/silver-walls-shake.md:1]: The changeset is missing some of the changed packages.

File: packages/keystone/tests/List.test.js [hunk 10]
```diff
@@ -142,15 +142,17 @@ describe('new List()', () => {
...
-      public: {
-        create: true,
-        delete: true,
-        read: true,
-        update: true,
-        auth: true,
-      },
-    });
+    expect(list.access).toEqual(
+      expect.objectContaining({
```
Comment (timleslie) [inline packages/keystone/tests/List.test.js:146]: Let's update this to explicitly contain the `internal` values as well.

**Pull request merged**